src/mechanics/IGA/NURBS.h

```diff
@@ -0,0 +1,478 @@
+#pragma once
+
+#include<eigen3/Eigen/Dense>
+#include <array>
+#include <vector>
+#include <iostream>
+#include "mechanics/nodes/NodeSimple.h"
+
+namespace NuTo
+{
+template <int TDimParameter>
+class NURBS
+{
+public:
+
+    enum mParametrization {chord, centripetal};
+
+    /** Constructors **/
+
+    //! @brief ... default constructor
+    NURBS(){}
+
+    //! @brief ... constructor
+    //! @param rDegree ... degree of the polynomial
+    //! @param rKnots ... knot vector
+    //! @param rControlPoints ... control points
+    NURBS(const std::array<std::vector<double>, TDimParameter> &rKnots,
```

Comment:
There has been an agreement that `const` references should no longer feature the prefix `r`.